biomolecules/geom/sample_geom_drugs.py

```diff
@@ -0,0 +1,86 @@
+import os
+import json
+import pickle
+import random
+import argparse
+import tqdm
+
+
+def write_pickle(data, path):
+    with open(path, "wb") as f:
+        pickle.dump(data, f, pickle.HIGHEST_PROTOCOL)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--base_path",
+        type=str,
+        required=True,
+        help="Path to where you untarred the rdkit folder i.e. path/to/rdkit_folder",
+    )
+    parser.add_argument(
+        "--out_path", type=str, required=True, help="Path to the output directory"
+    )
+
+    return parser.parse_args()
+
+
+def main():
+
+    args = parse_args()
+
+    # rdkit folder downloaded 2024-03-01
+    base_path = args.base_path
+    out_path = args.out_path
+    drugs_file = os.path.join(base_path, "summary_drugs.json")
+
+    with open(drugs_file, "r") as f:
+        drugs_summ = json.load(f)
+
+    # set random seed
+    random.seed(42)
+
+    train_id_list = []
+    val_id_list = []
+    id_mol_dict = {}
+    missing_drugs = []
+
+    for drug in tqdm.tqdm(list(drugs_summ.keys())[:1000]):
+        # not all drugs have necessary keys
+        try:
+            charge = drugs_summ[drug]["charge"]
+            pickle_path = drugs_summ[drug]["pickle_path"]
+        except KeyError:
+            missing_drugs.append(drug)
+            continue
+        pp = os.path.join(base_path, pickle_path)
+        with open(pp, "rb") as f:
+            mol_dict = pickle.load(f)
+        rand = random.random()
+        # generate train ids with 85% probability - ~5% charged
+        for i, c in enumerate(mol_dict["conformers"]):
+            id_mol_dict[c["geom_id"]] = {
+                "pickle_path": pickle_path,
+                "conf_idx": i,
+                "charge": charge,
+                "molecule": drug,
+            }
+            if charge == 0 and rand > 0.15:
+               train_id_list.append(c["geom_id"])
+            # if charge != 0 or rand =< 0.15:
+            else:
+               val_id_list.append(c["geom_id"])
```

Comment:
Looks like train/val splits are done at a conformer level. Do we want to do this at a drug level (i.e. predicting molecule conformers for drugs we haven't seen before); that feels like the more OOD task.

Reply:
@mshuaibii, good catch. Yes, we should be sampling on drug/molecule level not conformer level. The previous suggested change, switched that behavior. I will revert.

Reply:
@mshuaibii I think the before Brandon just reverted was correct and the train/val split was happening at the drug level. The `rand` variable is updated in the drug loop, not the conformer loop.